src/Tools/dotnet-monitor/StartupLogging.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal sealed class StartupLogging : IDisposable
+    {
+        private const string WindowsLocalNamedPipePrefix = @"\\.\pipe\";
+
+        private readonly IHostApplicationLifetime _lifetime;
```

Comment:
What is the difference between our current startup logging (such as the warnings around no auth) vs. using IHostApplicationLifetime? Should the other log messages be moved here?

Reply:
I did this to stop polluting the Startup class with logging aspects that are unrelated to ASP.NET startup. The exist logging can be moved over here to keep the purely logging aspects separate from the ASP.NET startup aspects. Want me to do that in this PR?

Reply:
We can definitely save it for another PR.